src/java-interop/java-interop-logger.c

```diff
@@ -60,124 +56,6 @@ __android_log_vprint (int prio, const char* tag, const char* fmt, va_list ap)
 unsigned int log_categories;
 int gc_spew_enabled;
 
-static FILE*
-open_file (LogCategories category, const char *path, const char *override_dir, const char *filename)
-{
-	char *p = NULL;
-	FILE *f;
-
-	if (path && access (path, W_OK) < 0) {
-		log_warn (category, "Could not open path '%s' for logging (\"%s\"). Using '%s/%s' instead.",
-				path, strerror (errno), override_dir, filename);
-		path  = NULL;
-	}
-
-	if (!path) {
-		create_public_directory (override_dir);
-		p     = path_combine (override_dir, filename);
-		path  = p;
-	}
-
-	unlink (path);
-
-	f = monodroid_fopen (path, "a");
-
-	if (f) {
-		set_world_accessable (path);
-	} else {
-		log_warn (category, "Could not open path '%s' for logging: %s",
-				path, strerror (errno));
-	}
-
-	free (p);
-
-	return f;
-}
-
-void
-init_categories (const char *override_dir)
-{
-	char *value;
-	char **args, **ptr;
-
-	const char *gref_file = NULL;
-	const char *lref_file = NULL;
-
-#if !ANDROID
-	log_categories = LOG_DEFAULT;
-#endif
-	int light_gref  = 0;
-	int light_lref  = 0;
-
-	if (monodroid_get_namespaced_system_property (DEBUG_MONO_LOG_PROPERTY, &value) == 0)
```

Comment:
Additionally, I'm somewhat *confused*: this `init_categories()` method reads the `debug.mono.log` Android system property to see what should be logged.  I do not see *any* mention of `debug.mono.log` here: https://github.com/mono/mono/commits/2018-04-sdks-android-standalone

I assume it's no longer "necessary," as `MONO_LOG_LEVEL` or `MONO_LOG_MASK` should be used instead, but *that change must be documented somewhere*.  At present, there's no docs about *anything*, as far as I can tell.

Which brings me back to the [original (badly made?) point](
https://github.com/xamarin/java.interop/pull/363#issuecomment-416281789): Is there a description, *somewhere* of what the `mono:2018-04-sdks-android-standalone` branch *does*?  Or are we just willy-nilly writing code all over the place to "do things," without any form of cohesive plan or description?